org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/LSSymbolsContentProvider.java

```diff
@@ -144,6 +150,39 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 		public void documentChanged(DocumentEvent event) {
 			refreshTreeContentFromLS();
 		}
+
+	}
+
+	private final class PreferencesChangedOutlineUpdater implements IPreferenceChangeListener, IOutlineUpdater {
+
+		@Override
+		public void install() {
+			IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(LanguageServerPlugin.PLUGIN_ID);
+			preferences.addPreferenceChangeListener(this);
+		}
+
+		@Override
+		public void uninstall() {
+			IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(LanguageServerPlugin.PLUGIN_ID);
+			preferences.removePreferenceChangeListener(this);
+		}
+
+		@Override
+		public void preferenceChange(PreferenceChangeEvent event) {
+			if (viewer == null || outlineViewerInput == null) {
+				return;
+			}
+
+			if (event.getKey().startsWith(CNFOutlinePage.HIDE_DOCUMENT_SYMBOL_KIND_PREFERENCE_PREFIX)) {
```

Comment:
Since `outlineViewerInput` is not used in this method, could we remove lines 172-175 and just add `viewer != null` to the `if`?

Note: I think that to avoid NullPointer warnings, you need assign `viewer` to a local such as `theViewer` and use that local in the rest of the method.

Reply:
I'll change that as suggested. Originally I used that variable here, but changed the code later and forgot to remove that dependency. Thanks.